lib/adbc_connection.ex

```diff
@@ -402,18 +496,32 @@ defmodule Adbc.Connection do
     end
   end
 
+  defp handle_stream({:query, query_or_prepared, params, statement_options}, conn) do
```

Comment:
I think we can delete the clause above now, no?

Also, should we allow `statement_options` as optional argument to `prepare`?

Reply:
Ah I finally found the above one which was used in `query_pointer`, and I've added `statement_options` to `query_pointer`.